dev/com.ibm.ws.classloading/src/com/ibm/wsspi/classloading/ApiType.java

```diff
@@ -61,15 +67,115 @@ public static ApiType fromString(String value) {
     /** Convert one or more comma-and-space-delimited api type strings into a single set of types */
     public static EnumSet<ApiType> createApiTypeSet(String... apiTypes) {
         EnumSet<ApiType> set = EnumSet.noneOf(ApiType.class);
+        EnumSet<ApiType> removeSet = EnumSet.noneOf(ApiType.class);
+        EnumSet<ApiType> addSet = EnumSet.noneOf(ApiType.class);
+        ArrayList<String> notFoundSet = new ArrayList<String>();
+        StringBuffer initialtypes = new StringBuffer();
         if (apiTypes != null)
-            for (String types : apiTypes)
+            for (String types : apiTypes) {
+                if (initialtypes.length() == 0)
+                    initialtypes.append("\"" + types + "\"");
+                else
+                    initialtypes.append(" \"" + types + "\"");
                 if (types != null)
                     for (String stype : types.split("[ ,]+")) {
-                        ApiType type = ApiType.fromString(stype);
-                        if (type != null) {
-                            set.add(type);
+                        if (stype.indexOf("+") == 0) {
+                            stype = stype.replaceFirst("\\+", "");
+                            ApiType type = ApiType.fromString(stype);
+                            if (type != null) {
+                                if (!addSet.add(type)) {
+                                    if (tc.isErrorEnabled())
+                                        Tr.error(tc, "cls.classloader.config.duplicate", stype, initialtypes);
+                                    set.clear();
+                                    return set;
```

Comment:
Can we throw an exception in the case of an error instead of returning an empty set?  It would be ideal if we could blow up completely on this code path, rather than letting the code continue on and getting a more confusing hard-failure later on.  It would be good to see what happens in a FAT test.

Reply:
No, I think that would likely change the behavior too much for this code,  I really think the errors and returning () will work since they will likely get class loading exceptions or error.   I could look into this under another pull next week if you really think its required to throw and exception.